src/libsyncengine/db/syncdb.cpp

```diff
@@ -1574,67 +1430,41 @@ bool SyncDb::parent(ReplicaSide side, const NodeId &nodeId, NodeId &parentNodeid
-        return false;
-    }
-    if (!found) {
-        return true;
-    }
-
-    // Fill names' vector
+    // names' vector
     std::vector<SyncName> names;
 
-    bool parentNodeDbIdIsNull;
-    DbNodeId parentNodeDbId;
-    ASSERT(queryIsNullValue(id, SELECT_NODE_BY_REPLICAID_PARENTID, parentNodeDbIdIsNull));
-    if (!parentNodeDbIdIsNull) {
-        ASSERT(queryInt64Value(id, SELECT_NODE_BY_REPLICAID_PARENTID, parentNodeDbId));
+    std::string requestId =
+            (side == ReplicaSide::Local ? SELECT_ANCESTORS_NODES_LOCAL_REQUEST_ID : SELECT_ANCESTORS_NODES_DRIVE_REQUEST_ID);
 
-        SyncName name;
-        ASSERT(querySyncNameValue(
-                id, side == ReplicaSide::Local ? SELECT_NODE_BY_REPLICAID_NAMELOCAL : SELECT_NODE_BY_REPLICAID_NAMEDRIVE, name));
-        names.push_back(name);
+    ASSERT(queryResetAndClearBindings(requestId));
+    ASSERT(queryBindValue(requestId, 1, nodeId));
 
-        ASSERT(queryResetAndClearBindings(id));
+    for (;;) {
+        bool hasNext = false;
+        if (!queryNext(requestId, hasNext)) {
+            LOG_WARN(_logger, "Error getting query result: " << requestId.c_str());
+            return false;
+        }
+        if (!hasNext) {
+            break;
+        }
 
-        while (!parentNodeDbIdIsNull) {
-            ASSERT(queryResetAndClearBindings(SELECT_NODE_BY_NODEID_LITE_ID));
-            ASSERT(queryBindValue(SELECT_NODE_BY_NODEID_LITE_ID, 1, parentNodeDbId));
-            if (!queryNext(SELECT_NODE_BY_NODEID_LITE_ID, found)) {
-                LOG_WARN(_logger,
-                         "Error getting query result: " << SELECT_NODE_BY_NODEID_LITE_ID
-                                                        << (side == ReplicaSide::Local ? " - nodeIdLocal=" : " - nodeIdDrive=")
-                                                        << nodeId.c_str());
-                return false;
-            }
-            if (!found) {
-                return true;
-            }
+        SyncName name;
+        ASSERT(querySyncNameValue(requestId, side == ReplicaSide::Local ? 0 : 1, name));
 
-            ASSERT(queryIsNullValue(SELECT_NODE_BY_NODEID_LITE_ID, 0, parentNodeDbIdIsNull));
-            if (!parentNodeDbIdIsNull) {
-                ASSERT(queryInt64Value(SELECT_NODE_BY_NODEID_LITE_ID, 0, parentNodeDbId));
+        names.push_back(name);
+    }
 
-                ASSERT(querySyncNameValue(SELECT_NODE_BY_NODEID_LITE_ID, (side == ReplicaSide::Local ? 1 : 2), name));
-                names.push_back(name);
-            }
+    ASSERT(queryResetAndClearBindings(requestId));
 
-            ASSERT(queryResetAndClearBindings(SELECT_NODE_BY_NODEID_LITE_ID));
-        }
-    } else {
-        ASSERT(queryResetAndClearBindings(id));
-    }
+    found = (names.size() > 0);
 
-    // Construct path from names' vector
-    path.clear();
-    for (auto nameIt = names.rbegin(); nameIt != names.rend(); ++nameIt) {
-        path.append(*nameIt);
+    if (found) {
```

Comment:
Same remarks as above: I would not test the `found` value and I would alway clear `path`.

Reply:
Done